frontend/src/metabase/query_builder/components/filters/FilterPopover/FilterPopover.tsx

```diff
@@ -242,83 +246,103 @@ export function FilterPopover({
   const supportsExpressions = query.database()?.supportsExpressions();
 
   const filterOperator = filter.operator();
-  const hasPicker = filterOperator && filterOperator.fields.length > 0;
+  const hasPicker =
+    (filterOperator && filterOperator.fields.length > 0) || isBoolean(field);
+  const stageIndex = -1; // this will not always be true
```

Comment:
this works... not sure why.

Reply:
Do you mean the `-1` value? It's a shortcut for "the last query stage". But yeah, we shouldn't be using it here